Source/Core/Core/AchievementManager.cpp

```diff
@@ -286,6 +291,251 @@ void AchievementManager::ActivateDeactivateRichPresence()
   INFO_LOG_FMT(ACHIEVEMENTS, "Rich presence (de)activated.");
 }
 
+void AchievementManager::FetchBadges()
+{
+  if (!m_is_runtime_initialized || !IsLoggedIn() || !Config::Get(Config::RA_BADGES_ENABLED))
+  {
+    if (m_update_callback)
+      m_update_callback();
+    return;
+  }
+  m_image_queue.Cancel();
+
+  if (m_player_badge.name != m_display_name)
+  {
+    m_image_queue.EmplaceItem([this] {
+      std::string name_to_fetch;
+      {
+        std::lock_guard lg{m_lock};
+        name_to_fetch = m_display_name;
+        if (name_to_fetch == m_player_badge.name)
+          return;
```

Comment:
You can invert this to avoid a copy in the equals case.

```
        if (m_player_badge.name == m_display_name)
          return;
        name_to_fetch = m_display_name;
```